collector/src/main/java/com/bitmovin/analytics/BitmovinAdAnalytics.kt

```diff
@@ -0,0 +1,226 @@
+package com.bitmovin.analytics
+
+import com.bitmovin.analytics.ads.*
+import com.bitmovin.analytics.data.AdSample
+import com.bitmovin.analytics.data.AdEventData
+import com.bitmovin.analytics.utils.Util
+
+class BitmovinAdAnalytics(var analytics: BitmovinAnalytics) {
+    private var activeAdBreak: AdBreak? = null
+    private var activeAdSample: AdSample? = null
+    private var adPodPosition: Int = 0
+    private var adStartupTimestamp: Long? = null
+    private var currentTime: Long? = null
+    private var beginPlayingTimestamp: Long? = null
+    private var isPlaying: Boolean = false
+    private val adManifestDownloadTimes: HashMap<String, Long> = hashMapOf()
+
+    fun onAdStarted(ad: Ad) {
+        if (!ad.isLinear) {
+            return
+        }
+
+        this.resetActiveAd()
+        this.activeAdSample = AdSample(ad = ad)
+        val activeAdSample = this.activeAdSample ?: return
+        this.currentTime = null
+        activeAdSample.adStartupTime = if (this.adStartupTimestamp != null) Util.getTimeStamp() - this.adStartupTimestamp!! else null
+
+        this.startAd(activeAdSample)
+    }
+
+    fun onAdFinished() {
+        val activeAdSample = this.activeAdSample ?: return
+        val activeAdBreak = this.activeAdBreak ?: return
+
+        val adSample = activeAdSample.copy()
+        adSample.completed = 1
+        this.resetActiveAd()
+        this.completeAd(activeAdBreak, adSample, adSample.ad.duration)
+    }
+
+    fun onAdBreakStarted(adBreak: AdBreak) {
+        this.adPodPosition = 0
+        this.activeAdBreak = adBreak
+        this.adStartupTimestamp = Util.getTimeStamp()
+    }
+
+    fun onAdBreakFinished() {
+        this.resetActiveAd()
+        this.activeAdBreak = null
+    }
+
+    fun onAdClicked(clickThroughUrl: String?) {
+        val activeAdSample = this.activeAdSample ?: return
+
+        activeAdSample.ad.clickThroughUrl = clickThroughUrl
+        activeAdSample.clicked = 1
+
+        val timestamp = Util.getTimeStamp()
+        val timePlayed: Long = if (activeAdSample.timePlayed != null) activeAdSample.timePlayed!! else 0
+        activeAdSample.clickPosition = timePlayed + timestamp - this.beginPlayingTimestamp!!
```

Comment:
I think we might be able to move all this current time calculation to some method or object inside the ad adapter. Feels like this is scattered all over :)